### ISSUE TITLE
Associations with REST routes not working on Windows

### ISSUE THREAD

**Issue by cyrilchapon** [balderdashy/sails#2787]
Hello,

When I try a "[REST-populate](http://sailsjs.org/#!/documentation/reference/blueprint-api/Populate.html)", or a "[REST-addTo](http://sailsjs.org/#!/documentation/reference/blueprint-api/Add.html)", that just doesn't work :

**api/models/Pet.js**

```
module.exports = {
  attributes: {
    name:'STRING',
    color:'STRING',
    owner:{
      model:'user'
    }
  }
};
```

**api/models/User.js**

```
module.exports = {
  attributes: {
    name:'STRING',
    age:'INTEGER',
    pets:{
      collection: 'pet',
      via: 'owner'
    }
  }
};
```

`GET /user/1/pets` is returning **404**, with an existing *1* user
`POST /user/1/pets` with `{"name":"Medor","color":"Fire"}` is returning **404**, with an existing *1* user

I should add that shortcut routes do work :
`GET /user/1?populate=pets` works (*200*)
`GET /user/1/pets/add?name=Medor&color=Fire` works (*200*)

**Comment by zeluis9** [comment]
Working here. Which version of sails are you using?

**Comment by cyrilchapon** [comment]
@zeluis9 Apparently *0.11.0* both 

- globally

```
F:\nodejs\www>npm list sails
F:\nodejs
└── sails@0.11.0
```

- and locally

```
F:\nodejs\www\myproject>npm list sails
myproject@0.0.0 F:\nodejs\www\myproject
└── sails@0.11.0
```

**Comment by zeluis9** [comment]
My version is 0.10.5. Must be a bug in the newer version. In 0.10.5 works like a charm

**Comment by tjwebb** [comment]
Works fine for me in 0.11.0. Post response of `GET /user/1`

**Comment by cyrilchapon** [comment]
- `GET http://localhost:1337/user/1`

```
{
  "pets": [
    {
      "name": "Medor",
      "owner": 1,
      "id": 1,
      "createdAt": "2015-03-26T09:55:38.800Z",
      "updatedAt": "2015-03-26T09:55:38.832Z"
    }
  ],
  "name": "Roger",
  "age": 50,
  "id": 1,
  "createdAt": "2015-03-26T09:43:51.390Z",
  "updatedAt": "2015-03-26T09:55:38.818Z"
}
```

- `GET http://localhost:1337/user/1/pets`

Getting a **404** without any body

**Comment by tjwebb** [comment]
What's your blueprints config?

**Comment by cyrilchapon** [comment]
Everything default (stripped informative comments here for readability)

```
module.exports.blueprints = {
  // actions: true,
  // rest: true,
  // shortcuts: true,
  // prefix: '',
  // pluralize: false,
  // populate: true,
  // autoWatch: true,
  // defaultLimit: 30
};
```

NB: Changing `populate` to `false` works, but doesn't change anything to my bug.

**Comment by cyrilchapon** [comment]
Tested exactly same code (with same config and ORM) under Debian and it's working as expected.

Need someone to reprocude on Windows.

**Comment by tjwebb** [comment]
Ok. sails doesn't have any UNIX-specific code, so I'm not sure what wold cause this. I'm closing until someone can demonstrate a failing unit test on Windows. Thanks

**Comment by gelevanog** [comment]
I have a similar problem on Windows 10.

**Comment by tjwebb** [comment]
I haven't used Windows since 2007 and didn't even know Windows 10 existed, so I'l probably not the best person to help out with this issue.

I'll re-open, maybe someone else can jump in. Anyone with Windows knowledge, please post any insight you might have, thanks.

**Comment by blah238** [comment]
Is this the same as https://github.com/balderdashy/sails/issues/2658?

**Comment by cyrilchapon** [comment]
@blah238, don't know but I'm guessing it has something to do with backslashes in paths, or case in model names maybe. Would instinctively search for an `if(!exist) return res.notFound()`.

**Comment by sylvainlap** [comment]
I've got exactly the same problem (under Windows 7).

**Comment by dukebigalow** [comment]
Yes, I can also confirm that issue under Windows 8.1
I just updated from "0.10.5" to "0.11.0", because of a hook that was not working for the old version, and the routes stopped working.
For now reverting to "0.11.0-rc5" is working.

**Comment by randallmeeker** [comment]
Just tested a fresh install and confirmed not working for associations blueprints routes.
Windows 7 SP3 
Sails 0.11 
Node 0.10.32 

Note: regular routes and automatic sub-directory controller routes work just fine.

**Comment by randallmeeker** [comment]
Clue
All the associated routes (the ones that don't work) are being built with a starting backslash
I'm digging around, but I don't know the router at all. 

```
info: Binding route ::  all /* (REQUEST HOOK: addMixins)
info: Binding route ::  all /* (RESPONSES HOOK: addResponseMethods)
info: Binding route ::  all /* (I18N HOOK: addLocalizationMethod)
info: Binding route ::  all /* (VIEWS HOOK: addResViewMethod)
info: Binding route ::   /* (CSRF HOOK: CSRF)
info: Binding route ::  all /* (CORS HOOK: clearHeaders)
info: Binding route ::   /csrfToken (CORS HOOK: sendHeaders)
info: Binding route ::  options /csrftoken (CORS HOOK: preflight)
info: Binding route ::   / (FUNCTION: rememberViewId)
info: Binding route ::   / (FUNCTION: serveView)
info: Binding route ::   /csrfToken (FUNCTION: csrfToken)
info: Binding route ::  get /__getcookie (SOCKETS HOOK)
info: Binding action (identity) blueprint/shadow route for controller: pet
info: Binding action (globalid) blueprint/shadow route for controller: pet
info: Binding action (sails) blueprint/shadow route for controller: pet
info: Binding route ::   /pet/find (POLICY: alwaysAllow)
info: Binding route ::   /pet/find (BLUEPRINT: findRecords)
info: Binding route ::   /pet/find/:id (POLICY: alwaysAllow)
info: Binding route ::   /pet/find/:id (BLUEPRINT: findOneRecord)
info: Binding route ::   /pet/create (POLICY: alwaysAllow)
info: Binding route ::   /pet/create (BLUEPRINT: createRecord)
info: Binding route ::   /pet/update/:id (POLICY: alwaysAllow)
info: Binding route ::   /pet/update/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::   /pet/destroy/:id? (POLICY: alwaysAllow)
info: Binding route ::   /pet/destroy/:id? (BLUEPRINT: destroyOneRecord)
info: Binding route ::  get /pet (POLICY: alwaysAllow)
info: Binding route ::  get /pet (BLUEPRINT: findRecords)
info: Binding route ::  get /pet/:id (POLICY: alwaysAllow)
info: Binding route ::  get /pet/:id (BLUEPRINT: findOneRecord)
info: Binding route ::  post /pet (POLICY: alwaysAllow)
info: Binding route ::  post /pet (BLUEPRINT: createRecord)
info: Binding route ::  put /pet/:id (POLICY: alwaysAllow)
info: Binding route ::  put /pet/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::  post /pet/:id (POLICY: alwaysAllow)
info: Binding route ::  post /pet/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::  delete /pet/:id? (POLICY: alwaysAllow)
info: Binding route ::  delete /pet/:id? (BLUEPRINT: destroyOneRecord)
info: Binding route ::  get \pet/:parentid/user/:id? (POLICY: alwaysAllow)
info: Binding route ::  get \pet/:parentid/user/:id? (BLUEPRINT: expand)
info: Binding action (identity) blueprint/shadow route for controller: user
info: Binding action (globalid) blueprint/shadow route for controller: user
info: Binding action (sails) blueprint/shadow route for controller: user
info: Binding route ::   /user/find (POLICY: alwaysAllow)
info: Binding route ::   /user/find (BLUEPRINT: findRecords)
info: Binding route ::   /user/find/:id (POLICY: alwaysAllow)
info: Binding route ::   /user/find/:id (BLUEPRINT: findOneRecord)
info: Binding route ::   /user/create (POLICY: alwaysAllow)
info: Binding route ::   /user/create (BLUEPRINT: createRecord)
info: Binding route ::   /user/update/:id (POLICY: alwaysAllow)
info: Binding route ::   /user/update/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::   /user/destroy/:id? (POLICY: alwaysAllow)
info: Binding route ::   /user/destroy/:id? (BLUEPRINT: destroyOneRecord)
info: Binding route ::   /user/:parentid/pets/add/:id? (POLICY: alwaysAllow)
info: Binding route ::   /user/:parentid/pets/add/:id? (BLUEPRINT: addToCollection)
info: Binding route ::   /user/:parentid/pets/remove/:id? (POLICY: alwaysAllow)
info: Binding route ::   /user/:parentid/pets/remove/:id? (BLUEPRINT: remove)
info: Binding route ::  get /user (POLICY: alwaysAllow)
info: Binding route ::  get /user (BLUEPRINT: findRecords)
info: Binding route ::  get /user/:id (POLICY: alwaysAllow)
info: Binding route ::  get /user/:id (BLUEPRINT: findOneRecord)
info: Binding route ::  post /user (POLICY: alwaysAllow)
info: Binding route ::  post /user (BLUEPRINT: createRecord)
info: Binding route ::  put /user/:id (POLICY: alwaysAllow)
info: Binding route ::  put /user/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::  post /user/:id (POLICY: alwaysAllow)
info: Binding route ::  post /user/:id (BLUEPRINT: updateOneRecord)
info: Binding route ::  delete /user/:id? (POLICY: alwaysAllow)
info: Binding route ::  delete /user/:id? (BLUEPRINT: destroyOneRecord)
info: Binding route ::  post \user/:parentid/pets/:id? (POLICY: alwaysAllow)
info: Binding route ::  post \user/:parentid/pets/:id? (BLUEPRINT: addToCollection)
info: Binding route ::  delete \user/:parentid/pets/:id? (POLICY: alwaysAllow)
info: Binding route ::  delete \user/:parentid/pets/:id? (BLUEPRINT: remove)
info: Binding route ::  get \user/:parentid/pets/:id? (POLICY: alwaysAllow)
info: Binding route ::  get \user/:parentid/pets/:id? (BLUEPRINT: expand)
```

**Comment by randallmeeker** [comment]
Ok, looks like it was fixed with this
https://github.com/balderdashy/sails/pull/2666

**Comment by sgress454** [comment]
Oy.  Always good to try searching through the issues before posting!

**Comment by randallmeeker** [comment]
@sgress454 I did, and always do. That particular issue really had no keywords that addressed this issue and it took time just find out that that issue addressed it. I was attempting to make a PR on master only when I found that someone else had.

**Comment by cyrilchapon** [comment]
> @sgress454 I did, and always do.

And so I

**Comment by gfaceless** [comment]
I didn't know it is a Windows-specific issue until I found this post.
should have used the latest version :(

**Comment by YMA-MDL** [comment]
did this issue come back in 0.11.2?  because I doesn't work for me (Win 7 Pro)

**Comment by davestevens** [comment]
I've just hit this issue too. Looks like the fix (#2666) first appears in v0.12.0-rc1

**Comment by davestevens** [comment]
Running `npm install sails@0.12.0-rc1` fixes the issue and my blueprint routes are now bound to `/api/` rather than `\api\`.
